docs/operational_guide/availability_consistency_durability.md

```diff
@@ -0,0 +1,71 @@
+# Tuning Availability, Consistency, and Durability
+
+## Overview
+
+Generally speaking M3DB is an [H.A](https://en.wikipedia.org/wiki/High_availability) system because it doesn't use a consensus protocol like Raft or Paxos to enforce strong consensus and consistency guarantees. However, even within the category of H.A systems, there is a broad spectrum of consistency and durability guarantees that a database can provide. Like many other H.A systems, M3DB allows database operators to tune the consistency and durability guarantees of the system in order to achieve the desired balance between performance, availability, durability, and consistency.
+
+Generally speaking, all of the default and example YAML configuration files for M3DB are configured to favor performance and availability over consistency and durability since that is usually the tradeoff that makes most sense for metrics and observability workloads. Database operators who are using M3DB for workloads that require stricter consistency and durability guarantees should consider tuning the default configuration to better suite their use case. The rest of this document describes the various configuration options that are available to M3DB operators to make those tradeoffs.
```

Comment:
Yeah I'm happy to try and break it down more, the thing I struggle with though is that its all kinda mushed together.

For example, write/read consistency is mostly about consistency/durability, however, its also a performance thing. If you only require a write to succeed to one node, your P99 latency is likely to be much lower.

Similarly the async commitlog configuration is for performance reasons, but it has an impact on your durability and as a result, your consistency.

I agree that it feels like too much stuff mixed together but I'm also not sure how to separate them conceptually because they really are all tangled together

Reply:
yeah, i know what you mean. in my mind it looks something like this:

## Tuning For Performance & Availability
- async commitlog, incl consistency/durability caveats
- larger buffers for more throughput, incl consistency/durability caveats
- etc

## Tuning For Consistency & Durability
- sync commitlog, incl performance caveats
- smaller buffers for more flushing and lower margin for data loss, incl performance caveats
- etc

the gist is that yeah, you're right, they are intertwined - they're essentially inversely related concepts, so we do need to talk about everything in both sections. they're mutually exclusive, however, and each section is geared toward its half of the requirements. (and it's actually good that the sections overlap on subject matter - although not intent - because folks might come to the doc looking for their specific use case, and should have a summary of all the required information without needing to piece it together throughout the doc.)